# HG changeset patch
# User andrew
# Date 1378467880 -3600
# Node ID 3372405e49a9e5cd27af64e16eb7e44539ea3339
# Parent  b2611b1a750cf38b795d1df24c5bea9b9c246e0f
RH661505: JPEGs with sRGB IEC61966-2.1 color profiles have wrong colors

diff -r b2611b1a750c -r 3372405e49a9 src/share/native/sun/awt/image/jpeg/imageioJPEG.c
--- openjdk/jdk/src/share/native/sun/awt/image/jpeg/imageioJPEG.c	Wed Sep 04 18:20:38 2013 +0100
+++ openjdk/jdk/src/share/native/sun/awt/image/jpeg/imageioJPEG.c	Fri Sep 06 12:44:40 2013 +0100
@@ -1454,6 +1454,8 @@
                                                     "[S");
 }
 
+#define JPEG_APP1  (JPEG_APP0 + 1)  /* EXIF APP1 marker code  */
+
 JNIEXPORT jlong JNICALL
 Java_com_sun_imageio_plugins_jpeg_JPEGImageReader_initJPEGImageReader
     (JNIEnv *env,
@@ -1507,6 +1509,7 @@
 
     // Set up to keep any APP2 markers, as these might contain ICC profile
     // data
+    jpegsavemarkers(cinfo, JPEG_APP1, 0xFFFF);
     jpeg_save_markers(cinfo, ICC_MARKER, 0xFFFF);
 
     /*
@@ -1569,8 +1572,6 @@
     imageio_init_source((j_decompress_ptr) cinfo);
 }
 
-#define JPEG_APP1  (JPEG_APP0 + 1)  /* EXIF APP1 marker code  */
-
 /*
  * For EXIF images, the APP1 will appear immediately after the SOI,
  * so it's safe to only look at the first marker in the list.


